doc/tutorials/cmaes/cmaes.txt

```diff
@@ -0,0 +1,304 @@
+/*!
+
+@file cmaes.txt
+@author Kartik Nighania
+@brief Tutorial on how to use the CMAES optimizer class.
+
+@page cmaestutorial CMAES Optimizer tutorial (mlpack_cmaes)
+
+@section intro_cmaestut Introduction
+
+The CMA-ES (Evolution Strategy with Covariance Matrix Adaptation) is a robust 
+search/optimization method. The goal is to minimize a given objective 
+function, \f$f: R^N \rightarrow R\f$. The CMA-ES should be applied, if e.g. 
+BFGS and/or conjugate gradient methods fail due to a rugged search landscape 
+(e.g. discontinuities, outliers, noise, local optima, etc.). Learning the  
+covariance matrix in the CMA-ES is similar to learning the inverse Hessian matrix in 
+a quasi-Newton method. On smooth landscapes the CMA-ES is roughly ten times slower 
+than BFGS, assuming derivatives are not directly available. For up to \f$N=10\f$ 
+parameters the simplex direct search method (Nelder & Mead) is sometimes faster, but 
+less robust than CMA-ES.  On considerably hard problems the search (a single run) is 
+expected to take between \f$100\cdot N\f$ and \f$300\cdot N^2\f$ function evaluations. 
+
+@section toc_cmaestut Table of Contents
+
+A list of all the sections this tutorial contains.
+
+ - \ref intro_cmaestut
+ - \ref toc_cmaestut
+ - \ref cmaes_cmaestut
+   - \ref cmaes_ex1_cmaestut
+   - \ref cmaes_ex2_cmaestut
+   - \ref cmaes_ex3_cmaestut
+   - \ref cmaes_ex4_cmaestut
+ - \ref further_doc_cmaestut
+
+@section cmaes_cmaestut The 'CMAES' optimizer class
+
+The 'CMAES' class is a simple implementation of CMAES optimizer to converge a given 
+function or model.
+
+Using the CMAES class is very simple and can be divided into 3 simple steps- 
+
+1) The optimizer takes a function class which has the function described that user wants 
+to converge. 
+
+2) The CMAES object is made in which the constructor requires 6 input parameters out of 
+which only 1 parameter is compulsory.
```

Comment:
i have made a seperate section giving details about all constructor parameters and default values.
i will add the compulsory parameter here too..
thanks